### DISCUSSION TITLE
Error in milvus compilation

### DISCUSSION THREAD

**Discussion by luckercs** [milvus-io/milvus#28747] [Q&A and General discussion]
Verison: milvus 2.3.3
Docker compile

The error during compilation is as follows：
`
rocksdb/6.29.5: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/milvus/.conan/data/rocksdb/6.29.5/_/_/package/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/src"
-- Using Conan toolchain: /home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting BUILD_SHARED_LIBS = ON
-- The CXX compiler identification is GNU 9.4.0
-- The C compiler identification is GNU 9.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Conan: Component target declared 'zstd::libzstd_static'
-- Found Git: /usr/bin/git (found version "2.25.1") 
-- Performing Test HAVE_OMIT_LEAF_FRAME_POINTER
-- Performing Test HAVE_OMIT_LEAF_FRAME_POINTER - Success
-- Performing Test HAS_ARMV8_CRC
-- Performing Test HAS_ARMV8_CRC - Success
--  HAS_ARMV8_CRC yes
-- Performing Test BUILTIN_ATOMIC
-- Performing Test BUILTIN_ATOMIC - Success
-- Could NOT find uring (missing: uring_LIBRARIES uring_INCLUDE_DIR) 
-- Performing Test HAVE_THREAD_LOCAL
-- Performing Test HAVE_THREAD_LOCAL - Success
-- Disabling RTTI in Release builds
-- Performing Test HAVE_FALLOCATE
-- Performing Test HAVE_FALLOCATE - Success
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE
-- Performing Test HAVE_SYNC_FILE_RANGE_WRITE - Success
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP
-- Performing Test HAVE_PTHREAD_MUTEX_ADAPTIVE_NP - Success
-- Looking for malloc_usable_size
-- Looking for malloc_usable_size - found
-- Looking for sched_getcpu
-- Looking for sched_getcpu - found
-- Looking for getauxval
-- Looking for getauxval - not found
-- Looking for F_FULLFSYNC
-- Looking for F_FULLFSYNC - not found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- ROCKSDB_PLUGINS: 
-- JNI library is disabled
-- Configuring done
CMake Error at CMakeLists.txt:1038 (target_link_libraries):
  Target "rocksdb" links to:

    zstd::zstd

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

CMake Error at CMakeLists.txt:1043 (target_link_libraries):
  Target "rocksdb-shared" links to:

    zstd::zstd

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
rocksdb/6.29.5: 
rocksdb/6.29.5: ERROR: Package '8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d' build failed
rocksdb/6.29.5: WARN: Build folder /home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/build/Release
ERROR: rocksdb/6.29.5: Error in build() method, line 151
	cmake.configure()
	ConanException: Error 1 while executing cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/milvus/.conan/data/rocksdb/6.29.5/_/_/package/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/milvus/.conan/data/rocksdb/6.29.5/_/_/build/8f7abb7f330ae32d1c9c86ad0bffcf3379ba902d/src"
conan install failed
make: *** [Makefile:200: build-3rdparty] Error 1
`

**Comment by xiaofan-luan** [comment]
please cleanup conan dependency and this should be fixed already

**Reply by luckercs** [reply]
I have seen the relevant issue, thank you
